### ISSUE TITLE
Scale forever hulk with time instead of deaths, reward player for killing it

### ISSUE THREAD

**Issue by Terrorforge** [CleverRaven/Cataclysm-DDA#74958]
### Is your feature request related to a problem? Please describe.

Since discovering the new "Hunted Forever" scenario variant by @RenechCDDA (#73227) I've been playing with it a lot, specifically the mindhunter variant for MoM by @Standing-Storm (#73518). It's very cool, adding a neat new paranoid twist to the game, but I feel it has a critical flaw: you don't really get to engage with the power-up mechanic, because you have every incentive to never, ever kill the hulk. Even ignoring the fact that killing it just makes it more dangerous, it doesn't even get you a reprieve. When I escaped it and took a five-minute drive, it didn't catch up for the next 48 hours. When I killed it, it was back the next day.

This encourages players to just try and give it the slip forever, making it more of a nuisance than a threat and really neutering the whole "the only escape is death" angle. It also makes fortifying a position with turrets and mines a much less appealing solution, which would otherwise be an interesting dynamic and something many players feel is currently missing from the game.

I think players should be afraid of the hulk, but also _want_ to fight it. Killing it should feel like an accomplishment, not a mistake.

### Solution you would like.

The numbers are up for debate, but here are my suggestions:

1. Make the hulk scale with time instead of deaths. I suggest incrementing the buff counter by 1 every 5 days. Rebalance the Mindhunter to gain new powers every third tick instead of every second. By the end of summer it would have 118 speed, 39 armor across the board and (MoM) have just unlocked a dash and the ability to break your Concentration. That sounds about right to me.
2. When the player kills the hulk, make it go away for an extended period. I suggest 10 days, varying by 1-3 days in either direction just to keep the player on their toes.

### Describe alternatives you have considered.

Scale by time AND deaths. Don't particularly like this, because it puts us back to the point where you're encouraged not to fight the hulk, which seems like a shame. It's the most dangerous enemy in the game (except maybe the apeirohedron), I want to fight it.

### Additional context

I'm wondering if boosting its speed and conventional armor is enough to keep it threatening in the long run. In vanilla it takes a long time for it to boost its way out of range of higher-caliber guns that can snipe it from across the screen and it never stops being vulnerable to a well-placed grenade. In MoM, despite its enhanced offensive capabilities it feels like you can probably just nuke it since you have easy access to unconventional damage types like heat and photokinetic.

But I haven't actually played against high-tier hulks yet, so I'm not sure if its speed and tendency to ambush you in your sleep compensates for that.

**Comment by RenechCDDA** [comment]
I agree with your general idea that gameplay flow rewards avoiding it rather than killing it, but suggestion 2 is somewhat tricky to implement. I'm gonna think on it over the night and see what I can come up.

**Comment by Terrorforge** [comment]
Oh yeah, the nemesis functionality is hardcoded, isn't it? I was thinking you could do something like how MoM handles its power learning intervals.  E.g. create a NEMESIS_IS_DEAD variable set to 0 that ticks up to 1 when the nemesis dies, and then you have a recurring EOC that fires every 4-6 days that looks like:
`if NEMESIS_IS_DEAD=0
   then DO_NOTHING
else if NEMESIS_IS_DEAD=1
   then set NEMESIS_IS_DEAD=2
else if NEMESIS_IS_DEAD>=2
   then SPAWN_NEW_NEMESIS,
   set NEMESIS_IS_DEAD=0`
This would ensure the EOC doesn't happen to tick over and resurrect the nemesis two hours after you kill it. But I don't know if it's possible for an EOC to talk to the nemesis handling in a way that would make this workable.

**Comment by Terrorforge** [comment]
I killed it again, and just waited around in the same area for a while. This time, it took almost 5 full days before it showed up again. Does it just spawn in near your starting location and then start traveling towards you or something?

**Comment by Terrorforge** [comment]
Okay, so I did some code diving, and I think I figured it out. So, for future refernce:

When the current Nemesis dies, the game immediately starts the Nemesis mission again, which spawns another Nemesis and places it in a random "field" omt 40-80 omts away. Unless it can't, in which case it expands the search radius and tries again a couple of times before throwing an error.

Since EOCs can assign missions, implementing a delay shouldn't be too hard, right? Use a recurring EOC like I described, and if it fires when the Nemesis is dead, "assign_mission" : "kill_nemesis". Only complication is that the player will be able to tell if the Nemesis is back based on whether they have the mission or not. I would personally prefer if that weren't the case, because I think the unpredictable nature of the Nemesis is part of what makes it scary, but it's not a disaster.

**Comment by Terrorforge** [comment]
Some random thoughts after playing the game mode some more:

It's Summer 71, and I've pretty much forgotten I'm playing Mindhunter because I haven't seen the bastard since it punted my car into a pond on Summer 8. I moved around so much it fell massively behind; I've been stationary for three weeks, and it still hasn't caught up. I'm sure it will eventually, but I'm not remotely worried about it. I've only killed it 3 or 4 times, so it's pretty tame still. I've made no special preparation, because when it shows up I expect I'll be able to just shoot it to death with an assault rifle, or else mind-nuke it into dust before it knows what the fuck. At that point I guess I might start to move around more again just because I don't want to have to kill it every day whikle it smashes my workshop to pieces, but that's more of an annoyance than a threat.

Under my proposed system, I would be getting pretty scared of the Mindhunter by now, and would be making preparations for it. I would've kept moving, or else I would've found a more secure location to rest than just chucking a bed in the corner of a junkyard garage.

This confirms my initial feelings. In fact, I may have underestimated the problem. Running away is even more effective than I expected because the hulk moves so slowly across the map. I think constantly staying ahead of it should be an option, but then you should have to actually stay on the move, not burn rubber for a few days and then chill for a month. I'm not sure there's a good fix for that, though.
